### PR TITLE
S3 tests

### PR DESCRIPTION
**Pull request by nicolasbeauvais**
As discussed in #381 

I tough it would be a good thing to test the file integrity in https://github.com/spatie/laravel-medialibrary/compare/master...nicolasbeauvais:s3-tests?expand=1#diff-bae027168159c1221823e2acf336b0b0R93

but the s3 bucket needs an open policy so that Travis-ci can read the file with `file_get_content()` so it's commented for now.

### CODE REVIEW

**Comment by nicolasbeauvais** [comment]
The teardown method that cleaned all the files uploaded to s3 doesn't work :/ I will investigate tomorrow night !

**Comment by freekmurze** [comment]
The open policy has been set, all stored objects are now publicly accessible.

File: tests/TestCase.php [hunk 1]
```diff
@@ -160,4 +167,31 @@ private function setUpMorphMap()
...
+        $this->canTestS3 = ! (bool) array_search(false, $s3Configuration);
+
+        $app['config']->set('filesystems.disks.s3', $s3Configuration);
+        $app['config']->set(
+            'laravel-medialibrary.s3.domain',
+            'https://'.$s3Configuration['region'].'.amazonaws.com/laravel-medialibrary'
+        );
+
+        if ($this->canTestS3) {
+            register_shutdown_function(function () {
```
Comment (nicolasbeauvais) [inline tests/TestCase.php]: Couldn't find a way to call `Storage::` or `app('filesystem.disk')` in a ̀tearDownAfterCLass` method, using `register_shutdown_function` is the only working way that I found to clean the S3 bucket

**Comment by nicolasbeauvais** [comment]
Everything should work now :+1: But as you said my PR Travis build didn't have access to the S3 env vars

**Comment by freekmurze** [comment]
I'll try out the test by merging your PR!

**Pull request merged**